MaterialLib/SolidModels/SolidsBase.h

```diff
@@ -0,0 +1,87 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef MATERIALLIB_SOLIDMODELS_SOLIDBASE_H_
+#define MATERIALLIB_SOLIDMODELS_SOLIDBASE_H_
+
+#include <memory>
+
+#include "ProcessLib/Deformation/BMatrixPolicy.h"
+#include "ProcessLib/Parameter/Parameter.h"
+
+namespace MeshLib
+{
+class Element;
+}
+
+namespace Solids
+{
+template <int DisplacementDim>
+struct SolidsBase
+{
+    struct InternalVariables
```

Comment:
I'll add a comment along these lines:
"InternalVariables store material model specific state (other than sigma and eps), which are usually material history dependent."

The  class name comes, because this state is internal to the material model, but now this being visible in the interface, maybe MaterialState, or MaterialInternalState, or InternalState would be a better name.